Content/NPCs/Town/HunterNPC.cs

```diff
@@ -52,17 +42,58 @@ public override void SetDefaults()
 		NPC.aiStyle = NPCAIStyleID.Passive;
 		AnimationType = NPCID.BestiaryGirl;
 
-		NPC.TryEnableComponent<NPCDeathEffects>(
+		NPC.TryEnableComponent<NPCHitEffects>(
 			c =>
 			{
-				c.AddGore($"{PoTMod.ModName}/{Name}_0", 1);
-				c.AddGore($"{PoTMod.ModName}/{Name}_1", 2);
-				c.AddGore($"{PoTMod.ModName}/{Name}_2", 2);
+				c.AddGore(new NPCHitEffects.GoreSpawnParameters($"{PoTMod.ModName}/{Name}_0", 1, static npc => npc.life <= 0));
+				c.AddGore(new NPCHitEffects.GoreSpawnParameters($"{PoTMod.ModName}/{Name}_1", 1, static npc => npc.life <= 0));
+				c.AddGore(new NPCHitEffects.GoreSpawnParameters($"{PoTMod.ModName}/{Name}_2", 2, static npc => npc.life <= 0));
 				
-				c.AddDust(DustID.Blood, 20);
+				c.AddDust(new NPCHitEffects.DustSpawnParameters(DustID.Blood, 20));
+			}
+		);
+
+		NPC.TryEnableComponent<NPCTownDialogue>(
+			c =>
+			{
+				c.AddDialogue(new NPCTownDialogue.DialogueEntry($"Mods.{PoTMod.ModName}.NPCs.{Name}.Dialogue.Common0"));
+				c.AddDialogue(new NPCTownDialogue.DialogueEntry($"Mods.{PoTMod.ModName}.NPCs.{Name}.Dialogue.Common1"));
+				c.AddDialogue(new NPCTownDialogue.DialogueEntry($"Mods.{PoTMod.ModName}.NPCs.{Name}.Dialogue.Common2"));
```

Comment:
I feel like this use case may be common enough to merit an `AddDialogue` overload that also has an `int range` parameter, adding 0 through range automatically.

Reply:
Oh this is nice actually, I wanted to also make it use `npc.ModNPC.GetLocalization` but that'd make it non-functional for vanilla NPCs. Overall just not very broad and rather limited.